safehttp/response_writer.go

```diff
@@ -6,26 +6,30 @@ import (
 
 // ResponseWriter TODO
 type ResponseWriter struct {
-	d  Dispatcher
-	rw http.ResponseWriter
+	d      Dispatcher
+	rw     http.ResponseWriter
+	header Header
```

Comment:
Add a comment that having this field unexported is crucial for security. Otherwise one can easily overwrite the struct bypassing all our safety guarantees.